src/item.cpp

```diff
@@ -4932,7 +4955,20 @@ bool item::process_tool( player *carrier, const tripoint &pos )
         }
         make( tmp->revert_to );
         active = false;
+        return false;
+    }
+
+    // If tool must be worn outermost and this is nolonger the case shutdown the tool
+    if (has_flag("ACT_OUTERMOST") && ! is_worn_outermost(*carrier)) {
+        tmp->invoke( carrier != nullptr ? carrier : &g->u, this, pos );
```

Comment:
Rather than invoking and then checking for `revert_to`, the result of the invocation should be used.